src/compiler/checker.ts

```diff
@@ -32140,7 +32140,7 @@ namespace ts {
             if (catchClause) {
                 // Grammar checking
                 if (catchClause.variableDeclaration) {
-                    if (catchClause.variableDeclaration.type) {
+                    if (catchClause.variableDeclaration.type && tryGetTypeFromEffectiveTypeNode(catchClause.variableDeclaration) !== unknownType) {

```

Comment:
See <https://github.com/microsoft/TypeScript/issues/36775#issuecomment-636451488>

```suggestion
                    if (catchClause.variableDeclaration.type && tryGetTypeFromEffectiveTypeNode(catchClause.variableDeclaration) !== anyType && tryGetTypeFromEffectiveTypeNode(catchClause.variableDeclaration) !== unknownType) {
```